### PR TITLE
Bugfix FXIOS-9899 [Toolbar redesign] Always show navigation actions

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9899)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21710)

## :bulb: Description
- Always show navigation actions (back, forward, and (conditionally) data clearance) in the address toolbar when on the homepage

| Before | After |
| ------------- | ------------- |
| simulator_screenshot_584E418E-898A-471C-AFE9-5BC41C3D552A | Simulator Screenshot - iPhone 15 - 2024-09-06 at 12 13 05 |

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Review by cyndichin — APPROVED** [review]
thanks looks good, like that we are simplifying the logic here :) !
do you think its worth adding to the comments `(i.e. portrait) or `(i.e. landscape, iPad)` mode?

**Comment by MattLichtenstein** [comment]
> thanks looks good, like that we are simplifying the logic here :) ! do you think its worth adding to the comments `(i.e. portrait) or `(i.e. landscape, iPad)` mode?

@cyndichin I've restructured the conditional a bit and added a more descriptive comment

File: firefox-ios/Client/Frontend/Browser/Toolbars/Redux/AddressBarState.swift [hunk 1]
```diff
@@ -406,25 +406,21 @@ struct AddressBarState: StateType, Equatable {
     ) -> [ToolbarActionState] {
         var actions = [ToolbarActionState]()
 
+        // no actions needed for the address toolbar, which is only shown in iPhone landscape and iPad,
+        // when the navigation toolbar is showing (denoting we are not in iPhone landsacpe or iPad)
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Browser/Toolbars/Redux/AddressBarState.swift:410]: nit: 
```suggestion
        // Navigation actions are only added to the the address toolbar when there is no navigation toolbar shown, which occurs in iPhone landscape and iPad
```

**Comment by cyndichin** [comment]
> > thanks looks good, like that we are simplifying the logic here :) ! do you think its worth adding to the comments `(i.e. portrait) or `(i.e. landscape, iPad)` mode?
> 
> @cyndichin I've restructured the conditional a bit and added a more descriptive comment

great, even cleaner! 🔥  I added a nit suggestion for the comment.

**Comment by thatswinnie** [comment]
Unfortunately there is a bug. On iPad with stage manager on a small window the cancel button is not shown in the navigation toolbar when editing the url
Simulator Screenshot - iPad Air 11-inch (M2) - 2024-09-09 at 09 21 18

**Comment by MattLichtenstein** [comment]
@thatswinnie ah, good catch - had not considered stage manager, though seems like this issue would've affected any window size when the navigation toolbar was shown. should be fixed now!

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**